### PR TITLE
+ [e2e] new upgrade test cases for deletion check

### LINKED ISSUE #215: [e2e] [BUG] Additional virtual machine an images created during zero downtime upgrade should be deleted after completion
## What's the test to develop? Please describe

A clear and concise description of what the test you want to develop.

## Prerequisite and dependency of test

Any prerequisite environment and pre-condition required for this test. 
Provide test case dependency here if any.

- [ ] `item 1`

## Additional context

Add any other context or screenshots about the test request here.

related issue: harvester/harvester#1855

### PR DESCRIPTION
**Pull request by lanfon72**
## Changes
- **ADD** Test case to check VM created for upgrade been deleted.
- **ADD** Test case to check volume created for upgrade been deleted.
- **ADD** Test case to check image created for upgrade been deleted.

To close #215

### CODE REVIEW

**Comment by lanfon72** [comment]
As https://github.com/harvester/harvester/issues/3604#issuecomment-1525668882, new test case to check volume deleted will fail as expected.

File: harvester_e2e_tests/integration/test_upgrade.py [hunk 1]
```diff
@@ -1296,3 +1296,34 @@ def test_verify_crds_existed(self, api_client):
...
+        upgrade_vols = [vol for vol in data['data'] if 'upgrade' in vol['id']]
+
+        assert not upgrade_vols, (
+            "Upgrade related Volume still available:\n"
+            f"{upgrade_vols}"
+        )
+
+    @pytest.mark.dependency(depends=["any_nodes_upgrade"])
+    def test_upgrade_image_deleted(self, api_client):
+        code, data = api_client.vms.get(namespace='harvester-system')
```
[the remark this answers is not in the document]
Comment (lanfon72) [reply harvester_e2e_tests/integration/test_upgrade.py:1322]: yes, it looks I didn't push the correct version, should be `api_client.images.get(...)`

updated

File: harvester_e2e_tests/integration/test_upgrade.py [hunk 2]
```diff
@@ -1296,3 +1296,34 @@ def test_verify_crds_existed(self, api_client):
...
+        )
+
+    @pytest.mark.dependency(depends=["any_nodes_upgrade"])
+    def test_upgrade_image_deleted(self, api_client):
+        code, data = api_client.images.get(namespace='harvester-system')
+        upgrade_images = [image for image in data['items']
+                          if 'upgrade' in image['spec']['displayName']]
+
+        assert not upgrade_images, (
+            "Upgrade VM still available:\n"
```
Comment (khushboo-rancher) [inline harvester_e2e_tests/integration/test_upgrade.py:1327]: This should be "Upgrade related image(s) still available".

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]
A small change required, otherwise LGTM

File: harvester_e2e_tests/integration/test_upgrade.py (hunk 2 above)
Comment (lanfon72) [reply harvester_e2e_tests/integration/test_upgrade.py]: yes.. updated, my brain looks started vacation before :rofl:

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**